src/service/xhr-impl.js

```diff
@@ -158,6 +161,13 @@ export class Xhr {
         user().assert(false, `Response must contain the` +
             ` ${ALLOW_SOURCE_ORIGIN_HEADER} header`);
       }
+
+      // If the `AMP-Redirect-To` header is set. Redirect the user to the
+      // given URL.
+      const redirectTo = response.headers.get(REDIRECT_TO_HEADER);
+      if (redirectTo) {
+        this.win.top.location.href = redirectTo;
```

Comment:
Lets add `assertHttpsUrl` and document limitation to https. We also need to document that this must be an absolute URL.